ethers-providers/src/transports/quorum.rs

```diff
@@ -190,15 +191,67 @@ impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
 
         let mut numbers = vec![];
         let mut errors = vec![];
+
+        // exit once required quorum weight is achieved
+        let required_weight = self.quorum_weight();
+        let mut weight_achieved = 0;
+
+        // timestamp of when we started
+        let start = SystemTime::now();
```

Comment:
should we use `Instant::now()` instead of `SystemTime` throughout? I think it'd make a lot of the calculations around durations much easier, as the Instant::duration_since also doesn't return a Result